iroh-api/benches/add.rs

```diff
@@ -42,30 +45,40 @@ pub fn add_benchmark(c: &mut Criterion) {
                     rpc_client: rpc_client.clone(),
                     metrics: MetricsConfig::default(),
                 };
-                let (_task, rpc) = executor.block_on(async {
+                let (_task, client, resolver) = executor.block_on(async {
                     let store = Store::create(config).await.unwrap();
                     let task = executor.spawn(async move {
                         iroh_store::rpc::new(server_addr, store).await.unwrap()
                     });
                     // wait for a moment until the transport is setup
                     // TODO: signal this more clearly
                     tokio::time::sleep(std::time::Duration::from_millis(500)).await;
-                    let rpc = Client::new(rpc_client).await.unwrap();
-                    (task, rpc)
+                    let client = Client::new(rpc_client).await.unwrap();
+                    let content_loader = FullLoader::new(
+                        client.clone(),
+                        FullLoaderConfig {
+                            http_gateways: Vec::new(),
+                            indexer: None,
+                        },
+                    )
+                    .unwrap();
+                    let resolver = Resolver::new(content_loader);
+
+                    (task, client, resolver)
                 });
                 b.to_async(&executor).iter(|| {
-                    let rpc = rpc.clone();
+                    let api = Api::from_client_and_resolver(client.clone(), resolver.clone());
```

Comment:
I tried this first:
```rust
let (_task, api) = executor.block_on(async {
   ...
   let api = Api::from_config(api_config).await.unwrap();
   (task, api)
});
b.to_async(&executor).iter(|| {
   let api = api.clone();
});
```
But the complier said "no clone found in this scope", even those I derived `Clone` on `Api`, and tried implementing it manually.

Any thoughts on why I couldn't do this?

Reply:
you might need to do this

```rust
```

Reply:
also make sure you are not running into the mocks

Reply:
yea it was a mocks issue, I'm side stepping this by not cloning the `Api` struct directly, and cloning the `Resolver` and `Client` instead... should be refactored when we fix the mocks situation.